### PR TITLE
Update MICM wrapper for species unit conversions

### PR DESCRIPTION
**Pull request by boulderdaze**
Closes [#73](https://github.com/ESCOMP/atmospheric_physics/issues/73)

- Added two unit conversion functions i.e. `mol m-3`  -> `kg kg-1` and vise versa
- Updated the meta data
- Set up the test environment within the repository

### CODE REVIEW

File: musica/micm/micm.meta
Comment (nusbaume) [inline musica/micm/micm.meta:55]: This should be changed to "dry_air_density" (based off conversations in my current standard names PR):

```suggestion
  standard_name = dry_air_density
```

**Review by nusbaume — CHANGES_REQUESTED** [review]
@boulderdaze Looks good!  Just have some hopefully minor change requests.

Also I am happy to leave the final decision up to you, @mattldawson, and others, but I personally think we could incorporate the Github Action test you made in your fork into this repo.  I did request some changes that might require modifying the test itself, and its unclear what our final testing framework will look like, but all else being equal I would rather get some test coverage in now then wait for a "perfect" system later.  If you all agree then we can either add the test and Github action to this PR now or add it as a separate PR later (either way is fine with me).  Thanks!

File: doc/ChangeLog
Comment (boulderdaze) [reply doc/ChangeLog:7]: Yes, thank you for catching this. Fixed

File: musica/micm/micm.F90
Comment (boulderdaze) [reply musica/micm/micm.F90:104]: Adopted

File: musica/micm/micm.F90
Comment (boulderdaze) [reply musica/micm/micm.F90:156]: You are correct, fixed

File: musica/micm/micm.meta
Comment (boulderdaze) [reply musica/micm/micm.meta:55]: Done

Comment (boulderdaze) [reply musica/micm/micm.meta]: Hi @nusbaume, I learned that this PR has not been merged yet. I guess we could wait for your PR merged in because if not, CAM-SIMA would complain about `dry_air_density`.  I also think  `NamesNotInDictionary.txt` has to be regenerated. Could you confirm it?

File: musica/micm/micm.F90
Comment (nusbaume) [inline musica/micm/micm.F90:36]: I think there is a typo here (`creat` -> `create`).  Also would it be better to specifically say something like `MICM log file` instead of `other file`, or do we not know what the other file actually is yet?

File: test/musica/micm/test_micm_api.F90
Comment (nusbaume) [inline test/musica/micm/test_micm_api.F90:108]: Should we have a check after `micm_run` here that stops the program if the returned `errcode` is non-zero?

File: test/musica/micm/test_micm_api.F90
Comment (nusbaume) [inline test/musica/micm/test_micm_api.F90:116]: Should we check the `errcode` here after `micm_final` as well, just to be safe?

**Review by nusbaume — CHANGES_REQUESTED** [review]
Thanks for the new testing infrastructure @boulderdaze!  I did have some more questions and suggestions, but hopefully nothing that causes any major difficulties.

File: .github/workflows/test.yaml
Comment (K20shores) [reply .github/workflows/test.yaml:8]: This is a thing we've added in all of our other PRs. The intent is to only run the tests once when you have a PR. Otherwise, the tests run both on pushes when a PR is active and because it is a PR. If there's a better way, I'd like to know but this is the best we can figure out for now.

File: cmake/atmosphericphysicsConfig.cmake.in
Comment (K20shores) [inline cmake/atmosphericphysicsConfig.cmake.in:5]: I don't believe this file is necessary. This is only used to make an install target and atmospheric physics won't be installed with cmake as far as I'm aware

File: cmake/cmake_uninstall.cmake.in
Comment (K20shores) [inline cmake/cmake_uninstall.cmake.in:1]: I also think this isn't required for this pr

**Review by K20shores — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yaml
Comment (boulderdaze) [reply .github/workflows/test.yaml:10]: Done

File: .github/workflows/test.yaml
Comment (nusbaume) [reply .github/workflows/test.yaml:8]: @K20shores @boulderdaze if the issue is that the tests are running on PRs twice after every pushed commit then I am pretty sure you are running into the exact issue I described above about `push` and PR `synchronize` events triggering at the same time.  Might be worth trying the `On:` changes I listed above instead of the `if:` statement to see if that fixes that particular issue or not.

Of course if it doesn't solve the double-trigger problem then I am happy to leave this code as-is.

File: cmake/atmosphericphysicsConfig.cmake.in
Comment (boulderdaze) [reply cmake/atmosphericphysicsConfig.cmake.in:5]: This makes sense, removed.

File: cmake/cmake_uninstall.cmake.in
Comment (boulderdaze) [reply cmake/cmake_uninstall.cmake.in:1]: Yes, removed

File: musica/micm/micm.F90
Comment (boulderdaze) [reply musica/micm/micm.F90:36]: Thank you for catching the typo! I missed it. Yeah I was also wondering where the log messages are to be written. Is it possible to create `micm` only log file or would they be part of the other file, ie. atmospheric physics log?

**Review by mattldawson — APPROVED** [review]
looks great! I think having these tests started is going to be very helpful moving forward

**Review by K20shores — APPROVED** [review]

File: musica/micm/micm.F90
Comment (nusbaume) [reply musica/micm/micm.F90]: It should be possible to have `micm` create it's own log file, or have that log information included in the CAM-SIMA log file (or both).  Just let me know if you or the rest of the MUSICA team have a preference.

**Review by nusbaume — APPROVED** [review]
Everything looks great to me now, thanks again!

File: musica/micm/micm.F90
Comment (peverwhee) [inline musica/micm/micm.F90:37]: I think you can remove the write statements in init and run for the error messages - when you return, SIMA will handle printing the error messages so this will result in a duplicate error message.

File: test/musica/micm/test_micm_api.F90
Comment (peverwhee) [inline test/musica/micm/test_micm_api.F90:113]: I think instead of "solver failed to converge", it would be more helpful to print the errmsg returned from micm_run

**Review by peverwhee — APPROVED** [review]
Thanks Jiwon! A couple of small comments, but nothing that should hold up the PR.

File: musica/micm/micm.F90
Comment (peverwhee) [reply musica/micm/micm.F90]: thanks!

**Pull request merged**
